src/app/routes/groupRoutes.js

```diff
@@ -42,26 +42,29 @@ function proxyObj() {
         userResDecorator: function (proxyRes, proxyResData, req, res) {
             let resData = proxyResData.toString('utf8');
             try {
-                logger.info({ msg: 'proxyObj' });
-                let response = data.result.response;
+                const uri = 'learner/group'
+                const context = {
+                    env: telemtryEventConfig.URL[uri].env || 'group'
+                }
                 let data = JSON.parse(resData);
+                let response = data.result.response;
                 data.result.response = { id: '', rootOrgId: '' };
-                if (data.responseCode === 'OK') {
-                    data.result.response.id = response.id;
-                    data.result.response.rootOrgId = response.rootOrgId;
+                if (data.responseCode === 'OK' || data.responseCode === 200) {
+                    if (response && response.id) {
+                        data.result.response.id = response.id;
+                        data.result.response.rootOrgId = response.rootOrgId;
+                    }
+                    // generate success event log
+                    telemetryHelper.getTelemetryAPISuceessData(proxyResData, req, uri);
+                } else {
+                    // generate error event log
+                    const option = telemetry.getTelemetryAPIError(data, res, context);
+                    telemetryHelper.logApiErrorEventV2(req, option);
                 }
                 if (req.method === 'GET' && proxyRes.statusCode === 404 && (typeof data.message === 'string' && data.message.toLowerCase() === 'API not found with these values'.toLowerCase())) res.redirect('/')
                 else return proxyUtils.handleSessionExpiry(proxyRes, data, req, res, data);
             } catch (err) {
-                const uri = 'learner/group'
-                const context = {
-                    env: telemtryEventConfig.URL[uri].env || 'group'
-                }
-                const option = telemetry.getTelemetryAPIError(JSON.parse(resData),proxyRes, context);
-                if (option) {
-                telemetryHelper.logApiErrorEventV2(req, option);
-                }
-                logger.error({ msg: 'learner route : userResDecorator json parse error:', proxyResData })
```

Comment:
catch block has only the code or any functional failure right. if any API event is failed, then it will be in try block only, and there we have one more condition right to check the status…if we don’t use the condition also it will not go to catch block right?